File: 0225-implement-stack-using-queues/0225-implement-stack-using-queues.go
type MyStack struct {
    nums []int
}


func Constructor() MyStack {
    return MyStack{
        nums: make([]int, 0),
    }
}


func (this *MyStack) Push(x int)  {
    this.nums = append(this.nums, x)
}


func (this *MyStack) Pop() int {
    lastKey := len(this.nums) - 1
    lastVal := this.nums[lastKey]
    this.nums = this.nums[:lastKey]
    return lastVal
}


func (this *MyStack) Top() int {
    return this.nums[len(this.nums) - 1]
}


func (this *MyStack) Empty() bool {
    return len(this.nums) == 0
}


/**
 * Your MyStack object will be instantiated and called as such:
 * obj := Constructor();
 * obj.Push(x);
 * param_2 := obj.Pop();
 * param_3 := obj.Top();
 * param_4 := obj.Empty();
 */

File: 0225-implement-stack-using-queues/0225-implement-stack-using-queues.php
class MyStack {
    /**
     */
    function __construct(private $stack = []) {
        
    }
  
    /**
     * @param Integer $x
     * @return NULL
     */
    function push($x) {
        $this->stack[] = $x;
    }
  
    /**
     * @return Integer
     */
    function pop() {
        $lastKey = array_key_last($this->stack);
        $lastVal = $this->stack[$lastKey];
        unset($this->stack[$lastKey]);
        return $lastVal;
    }
  
    /**
     * @return Integer
     */
    function top() {
        return $this->stack[array_key_last($this->stack)];
    }
  
    /**
     * @return Boolean
     */
    function empty() {
        return empty($this->stack);
    }
}

/**
 * Your MyStack object will be instantiated and called as such:
 * $obj = MyStack();
 * $obj->push($x);
 * $ret_2 = $obj->pop();
 * $ret_3 = $obj->top();
 * $ret_4 = $obj->empty();
 */